File: lib/sort.rb
class Array

  def bubble_sort
    i = 0
     current_length = self.length
    while i < self.length do 
        swap_counter = 0
        j = 0
        k = 1
        while j < current_length-1 do 
            if self[j] > self[k]
                self[j], self[k] = self[k], self[j]
                swap_counter += 1
            end
            j += 1
            k += 1
        end
        current_length -= 1
        if swap_counter == 0
            return self
        end
        i += 1
    end
    return self
  end

end

File: sort.js
function bubbleSort(arr){
  var i = 0;
  var currentLength = arr.length;
  for (i; i < arr.length; i++) {
    var swapCounter = 0;
    var j = 0;
    var k = 1;
    for (j; j < currentLength - 1; j++) {
      var jelement = arr[j];
      if (arr[j] > arr[k]) {
        arr[j] = arr[k];
        arr[k] = jelement;
        swapCounter++;
      }
      k++;
      }
      currentLength--;
      if (swapCounter === 0) {
        return arr;
    }
  i++;
  }
  return arr;
}

// do not touch this
module.exports = bubbleSort;

